[quant] Add op support for linear_relu_dynamic_fp16

supriyar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #63826
* __->__ #63824
* #63820
* #63799

Summary:
Add a fused operator implementation that will work with the quantization fusion APIs.
Once FBGEMM FP16 kernel supports relu fusion natively we can remove the addition from the PT operator.

Test Plan:
python test/test_quantization.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30503514](https://our.internmc.facebook.com/intern/diff/D30503514)

supriyar committed:
[quant] Add op support for linear_relu_dynamic_fp16

Summary:
Add a fused operator implementation that will work with the quantization fusion APIs.
Once FBGEMM FP16 kernel supports relu fusion natively we can remove the addition from the PT operator.

Test Plan:
python test/test_quantization.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar committed:
Update on "[quant] Add op support for linear_relu_dynamic_fp16"


Summary:
Add a fused operator implementation that will work with the quantization fusion APIs.
Once FBGEMM FP16 kernel supports relu fusion natively we can remove the addition from the PT operator.

Test Plan:
python test/test_quantization.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar commented:
@supriyar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30503514).

supriyar committed:
Update on "[quant] Add op support for linear_relu_dynamic_fp16"


Summary:
Add a fused operator implementation that will work with the quantization fusion APIs.
Once FBGEMM FP16 kernel supports relu fusion natively we can remove the addition from the PT operator.

Test Plan:
python test/test_quantization.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30503514](https://our.internmc.facebook.com/intern/diff/D30503514)

[ghstack-poisoned]

jerryzh168 commented on test/quantization/core/test_quantized_op.py:
```diff
@@ -2778,6 +2778,36 @@ def test_qlinear_legacy(self, batch_size, input_channels, output_channels):
                          msg="torch.ops.quantized.fbgemm_linear_dynamic results are off")
 
+    @skipIfNoFBGEMM
+    @given(
```
should we use loops instead of hypoothsis here?

supriyar replied:
removed it :)

jerryzh168 reviewed (approved):
looks good, I think we should probably remove hypothesis

supriyar committed:
Update on "[quant] Add op support for linear_relu_dynamic_fp16"


Summary:
Add a fused operator implementation that will work with the quantization fusion APIs.
Once FBGEMM FP16 kernel supports relu fusion natively we can remove the addition from the PT operator.

Test Plan:
python test/test_quantization.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30503514](https://our.internmc.facebook.com/intern/diff/D30503514)

[ghstack-poisoned]

supriyar commented:
@supriyar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30503514).

supriyar committed:
Update on "[quant] Add op support for linear_relu_dynamic_fp16"


Summary:
Add a fused operator implementation that will work with the quantization fusion APIs.
Once FBGEMM FP16 kernel supports relu fusion natively we can remove the addition from the PT operator.

Test Plan:
python test/test_quantization.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30503514](https://our.internmc.facebook.com/intern/diff/D30503514)

[ghstack-poisoned]

supriyar committed:
Update on "[quant] Add op support for linear_relu_dynamic_fp16"


Summary:
Add a fused operator implementation that will work with the quantization fusion APIs.
Once FBGEMM FP16 kernel supports relu fusion natively we can remove the addition from the PT operator.

Test Plan:
python test/test_quantization.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30503514](https://our.internmc.facebook.com/intern/diff/D30503514)

[ghstack-poisoned]
